lib/RequestData.js

```diff
@@ -353,9 +358,11 @@ RequestData.prototype.set = function (key, value, formatFunction) {
       });
     }
     else {
-      var upsertVal = (isFunction(formatFunction)) ?
-        formatFunction(keypather.get(req, value) || value) :
-          (keypather.get(req, value) || value);
+      var val = (emptyString(value) || !isString(value)) ?
+        value : (keypather.get(req, value) || value); // keypather edgecase
```

Comment:
this would prevent you from setting `req.body` to `'body'`, but we don't really need that case, I suppose haha